gserver/src/format.ts

```diff
@@ -67,22 +79,28 @@ export function correctIndents(text, indent, settings: Settings) {
                 }
             }
             //Now we should find current line format
-            const format = findFormat(line, settings);
+            const format: ResolvedFormat = findFormat(line, settings);
+            if (format && format.symbol === 'Rule:') {
+                currentIndentation = defaultIndentation;
+            }
             let indentCount;
             if (~line.search(/^\s*$/)) { indentCount = 0; }
-            else if (typeof format === 'number') {
-                indentCount = format;
+            else if (format && typeof format.value === 'number') {
+                indentCount = format.value + currentIndentation;
             } else {
                 // Actually we could use 'relative' type of formatting for both - relative and unknown strings
                 // In future this behaviour could be reviewed
-                const nextLine = textArr.slice(i + 1).find(l => typeof findFormat(l, settings) === 'number');
+                const nextLine = textArr.slice(i + 1).find(l => typeof findIndentation(l, settings) === 'number');
                 if (nextLine) {
-                    const nextLineFormat = findFormat(nextLine, settings);
-                    indentCount = nextLineFormat === null ? defaultIndentation : nextLineFormat;
+                    const nextLineIndentation = findIndentation(nextLine, settings);
+                    indentCount = nextLineIndentation === null ? defaultIndentation : nextLineIndentation;
                 } else {
                     indentCount = defaultIndentation;
                 }
             }
+            if (format && format.symbol === 'Rule:') {
+                currentIndentation++;
```

Comment:
Looks like, that indentation should be increased by 'rule' value, not by 1.

Reply:
Agreed, will change